validator/validator_test.go

```diff
@@ -33,21 +33,88 @@ func TestHashesAreEqualForEqualMemberships(t *testing.T) {
 
 }
 
-func TestValidatorFromString(t *testing.T) {
-	addr := "t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy"
-	netAddr := "/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ"
-
-	v, err := NewValidatorFromString(fmt.Sprintf("%v@%v", addr, netAddr))
-	require.NoError(t, err)
+type validatorStrTest struct {
+	addr    string
+	netAddr string
+	weight  string
+	correct bool
+}
 
-	a, err := address.NewFromString(v.ID())
-	require.NoError(t, err)
-	require.Equal(t, a, v.Addr)
+var validatorStrTests = []validatorStrTest{
+	{
+		"t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy",
+		"/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ",
+		"1",
+		true,
+	},
+	{
+		"t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy",
+		"/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ",
+		"0",
+		true,
+	},
+	{"",
+		"/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ",
+		"0",
+		false,
+	},
+	{"",
+		"/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ",
+		"10",
+		false,
+	},
+	{"",
+		"/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ",
+		"",
+		false,
+	},
+	{
+		"t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy", "",
+		"0",
+		false,
+	},
+	{
+		"t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy", "",
+		"20",
+		false,
+	},
+	{
+		"", "",
+		"",
+		false,
+	},
+}
 
-	m, err := multiaddr.NewMultiaddr(v.NetAddr)
-	require.NoError(t, err)
-	require.Equal(t, m.String(), v.NetAddr)
+var incorrectValidatorStrings = []string{
```

Comment:
If we keep the 64bit integer parsing, let's add a test with a weight that does not fit into 64 bits.

Reply:
Fixed